src/main/java/com/worldmodelers/kafka/processors/java/ExampleStreamProcessor.java

```diff
@@ -42,17 +42,17 @@ public ExampleStreamProcessor( Properties properties ) {
     // to serialize/deserialize the message itself (in this case the custom ExampleStreamMessageSerde
     // defined in the messages package
     private Serde<String> stringSerdes = Serdes.String();
-    private Serde<ExampleStreamMessage> streamMessageSerdes =new ExampleStreamMessageSerde();
+    private Serde<ExampleStreamMessage> streamMessageSerdes = new ExampleStreamMessageSerde();
 
     // The Topology object, created by StreamsBuilder, defines the flow of messages from
     // an input topic to output topics. The mapValues method is where you define a transformation
     // on the consumed message, and is where you plug in your application's logic.
     private Topology buildStream() {
         StreamsBuilder builder = new StreamsBuilder();
-        builder.stream( "stream.in", Consumed.with(stringSerdes, streamMessageSerdes) ).mapValues( (message) -> {
+        builder.stream( "stream.in", Consumed.with( stringSerdes, streamMessageSerdes ) ).mapValues( ( message ) -> {
             message.getBreadcrumbs().add( "scala-kafka-streams" );
-            return new ExampleStreamMessage( message.getId(), message.getBreadcrumbs() );
-        } ).to( "stream.out", Produced.with(stringSerdes, streamMessageSerdes) );
+            return message;
```

Comment:
Java is different from Scala in that Scala values default to being immutable so the convention for updating them is to return a new object.

In Java, objects are mutable so it's possible to update them in line. Updating the object in place would look more conventional to a Java developer.

Reply:
@johnhungerford